### DISCUSSION TITLE
Registering PreventRequestsDuringMaintenance Middleware Laravel 11

### DISCUSSION THREAD

**Discussion by twd3** [statamic/cms#10605] [Q&A] [answered]
I'm following this tip: https://statamic.dev/tips/excluding-the-control-panel-from-maintenance-mode
I know in V4 this was registered in the Kernel. Would I need to register this in /app/bootstrap/app.php? I also have this same question when reading this tip: https://statamic.dev/tips/content-security-policy
Any help appreciated. Thanks!

**Comment by edalzell** [comment]
What have you tried?

**Comment by twd3** [comment]
Well I'm confused about if I even need to and I'm not sure the docs are up to date so I haven't tried anything. I did create the Middleware folder and put that class in there. I also don't know about what I would put there to register it. Would it be this:
```$middleware->append(PreventRequestsDuringMaintenance::class);```
Or this:
```
$middleware->alias([
    'PreventRequestsDuringMaintenance' => \App\Http\Middleware\PreventRequestsDuringMaintenance::class::class,
]);
```

**Reply by NeoIsRecursive** [reply, marked as answer]
In `bootstrap/app.php` you should be able to do something like this in laravel 11 with the new skeleton:

```php
->withMiddleware(function (Middleware $middleware) {
    $middleware->preventRequestsDuringMaintenance(except: [
        '/cp*'
    ]);
});
```

If you have the old skeleton I think you just have to change the usage of the existing `PreventRequestsDuringMaintanance` with your own. In one of my projects I have done it like this in the `Http/Kernel` file:

```php
    protected $middleware = [
        ...
        \App\Http\Middleware\PreventRequestsDuringMaintenance::class,
        ...
    ];
```

**Reply by edalzell** [reply]
thanks @NeoIsRecursive, I wasn't sure what to recommend...now I know!

**Reply by duncanmcclean** [reply]
I've updated the docs to include the steps for excluding URLs in Laravel 11 apps.

Thanks for the solution, @NeoIsRecursive!

I've [updated the docs](https://statamic.dev/tips/excluding-the-control-panel-from-maintenance-mode#excluding-urls) to mention the different steps for Laravel 11 applications.

**Comment by twd3** [comment]
Oh so I don't even have to put that preventRequestsDuringMaintenance.php in the Middleware folder? This is on a fresh V5 install so it's the new skeleton...

**Reply by NeoIsRecursive** [reply]
Yep :) A lot of the middlewares that were in the kernel was moved and can be modified through the `Illuminate\Foundation\Configuration\Middleware` instance you get from `->withMiddleware` instead.

https://laravel.com/docs/11.x/releases#middleware

**Reply by twd3** [reply]
Yup, just deleted that file in Middleware and tried below. Works great! Thank you!
```
    $middleware->preventRequestsDuringMaintenance(except: [
        '/cp*'
    ]);
```
